### PR TITLE
Fix no export subcmd panic on mac

### PR DESCRIPTION
**Pull request by An-DJ**
## Relevant Issue (if applicable)
_If there are Issues related to this PullRequest, please list it._

Fix panic caused by no `export` subcmd on mac in issue #1340 

## Details
_Please describe the details of PullRequest._

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.

### CODE REVIEW

File: src/bin/nydus-image/main.rs [hunk 1]
```diff
@@ -773,8 +771,18 @@ fn main() -> Result<()> {
...
+        #[cfg(target_os = "linux")]
+        if let Some(matches) = cmd.subcommand_matches("export") {
+            Command::export(&cmd, matches, &build_info)
+        } else {
+            println!("{}", usage);
+            Ok(())
+        }
+        #[cfg(not(target_os = "linux"))]
+        {
+            println!("{}", usage);
```
Comment (imeoer) [inline src/bin/nydus-image/main.rs:783]: Thanks for the fixup! Maybe give the warning to tell users we can't support `export` subcommand for macos here?

Comment (An-DJ) [reply src/bin/nydus-image/main.rs:783]: Actually if the user run `export` on macos, it cannot be recognized.

```shell
% ./target/debug/nydus-image export
error: The subcommand 'export' wasn't recognized

Usage: nydus-image [OPTIONS] [COMMAND]

For more information try '--help'
```

IMHO this error msg is enough, isn't it ? :)

Comment (An-DJ) [reply src/bin/nydus-image/main.rs:783]: Also `export` has been removed from the available subcmd on macos:

```shell
% ./target/debug/nydus-image -h
Build, analyze, inspect or validate RAFS filesystems/Nydus accelerated container images

Usage: nydus-image [OPTIONS] [COMMAND]

Commands:
  create     Create RAFS filesystems from directories, tar files or OCI images
  chunkdict  deduplicate RAFS filesystem metadata
  merge      Merge multiple bootstraps into a overlaid bootstrap
  check      Validate RAFS filesystem metadata
  inspect    Inspect RAFS filesystem metadata in interactive or request mode
  stat       Generate statistics information for RAFS filesystems
  compact    (experimental)Compact specific nydus image, remove unused chunks in blobs, merge small blobs
  unpack     Unpack a RAFS filesystem to a tar file
  help       Print this message or the help of the given subcommand(s)

Options:
  -L, --log-file <log-file>    Log file path
  -l, --log-level <log-level>  Log level: [default: info] [possible values: trace, debug, info, warn, error]
  -h, --help                   Print help information
  -V, --version                Print version information

```

Comment (imeoer) [reply src/bin/nydus-image/main.rs:783]: Okay, maybe some macos users ask why not have `export` subcommand on his mac, but I think it's also okay for me.

**Review by imeoer — APPROVED** [review]

**Pull request merged**